File: Ser2023/Company/Company.Web/Modules/Company/PhoneNumberType/PhoneNumberTypeColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Company.Company.Columns;

[ColumnsScript("Company.PhoneNumberType")]
[BasedOnRow(typeof(PhoneNumberTypeRow), CheckNames = true)]
public class PhoneNumberTypeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public long PhoneNumberTypeId { get; set; }
    [EditLink]
    public string Name { get; set; }
    //public Guid Rowguid { get; set; }
    //public short IsActive { get; set; }
    //public int TenantId { get; set; }
}

File: Ser2025/Company/Company.Web/Modules/ServerTypes/Company/PhoneNumberTypeColumns.ts
import { ColumnsBase, fieldsProxy } from '@serenity-is/corelib';
import { Column } from '@serenity-is/sleekgrid';
import { PhoneNumberTypeRow } from './PhoneNumberTypeRow';

export interface PhoneNumberTypeColumns {
    PhoneNumberTypeId: Column<PhoneNumberTypeRow>;
    Name: Column<PhoneNumberTypeRow>;
    IsActive: Column<PhoneNumberTypeRow>;
    TenantId: Column<PhoneNumberTypeRow>;
}

export class PhoneNumberTypeColumns extends ColumnsBase<PhoneNumberTypeRow> {
    static readonly columnsKey = 'Company.PhoneNumberType';
    static readonly Fields = fieldsProxy<PhoneNumberTypeColumns>();
}